### PR TITLE
Add the gdal retile algorithm

### PR DESCRIPTION
**Pull request by havatv**
Goal:

Ticket(s): #

- [x] Backport to LTR documentation is required

### CODE REVIEW

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst [hunk 1]
```diff
@@ -753,6 +753,190 @@ Outputs
...
+------
+This utility will retile a set of input tile(s).
+All the input tile(s) must be georeferenced in the same
+coordinate system and have a matching number of bands.
+Optionally pyramid levels are generated.
+It is possible to generate shape file(s) for the tiled
+output.
+
+This algorithm is derived from the
+`GDAL Retile utility <https://gdal.org/gdal_retile.htmll>`_ .
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]:
```suggestion
`GDAL Retile utility <https://gdal.org/gdal_retile.html>`_ .
```

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst [hunk 2]
```diff
@@ -753,6 +753,190 @@ Outputs
      - The HTML file containing information about the input raster layer
 
 
+.. _gdalretile:
+
+Retile
+------
+This utility will retile a set of input tile(s).
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]:
```suggestion
Retiles a set of input tiles.
```
If it's a set then there are many, aren't they?

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst [hunk 3]
```diff
@@ -753,6 +753,190 @@ Outputs
...
+
+       Default: 256
+     - Width of the tiles in pixels (minimum 0)
+   * - **Tile height**
+     - ``TILE_SIZE_Y``
+     - [number]
+
+       Default: 256
+     - Height of the tiles in pixels (minimum 0)
+   * - **Overlap in pixels between consequtive tiles**
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]:
```suggestion
   * - **Overlap in pixels between consecutive tiles**
```

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst [hunk 4]
```diff
@@ -753,6 +753,190 @@ Outputs
...
+       * 4 --- Lanczos Windowed Sinc (``lanczos``)
+
+   * - **Column delimiter used in the CSV file**
+
+       Optional
+     - ``DELIMITER``
+     - [string]
+
+       Default: ';'
+     - CSV file delimiter
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]: I wondered what a csv file would be helpful for here, so maybe
```suggestion
     - Delimiter to use in the CSV file containing the tile(s) georeferencing information
```

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst [hunk 5]
```diff
@@ -753,6 +753,190 @@ Outputs
...
 
+.. _gdalretile:
+
+Retile
+------
+This utility will retile a set of input tile(s).
+All the input tile(s) must be georeferenced in the same
+coordinate system and have a matching number of bands.
+Optionally pyramid levels are generated.
+It is possible to generate shape file(s) for the tiled
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]: I can't see a parameter/output that addresses this. Do I miss anything?

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst [hunk 6]
```diff
@@ -753,6 +753,190 @@ Outputs
      - The HTML file containing information about the input raster layer
 
 
+.. _gdalretile:
+
+Retile
+------
+This utility will retile a set of input tile(s).
+All the input tile(s) must be georeferenced in the same
+coordinate system and have a matching number of bands.
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]: Kind of formulation I was suggesting for the raster calc

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst [hunk 7]
```diff
@@ -753,6 +753,190 @@ Outputs
...
+   :stub-columns: 0
+
+   * - Label
+     - Name
+     - Type
+     - Description
+   * - **Input files**
+     - ``INPUT``
+     - [raster] [list]
+     - The input raster files (can be multiple files).
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]: Do we need the comments in parenthesis (here and the next two)? Should be obvious.

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst [hunk 8]
```diff
@@ -753,6 +753,190 @@ Outputs
...
+   * - **Tile width**
+     - ``TILE_SIZE_X``
+     - [number]
+
+       Default: 256
+     - Width of the tiles in pixels (minimum 0)
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]: 0 is allowed?

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst [hunk 9]
```diff
@@ -753,6 +753,190 @@ Outputs
...
+     - Type
+     - Description
+   * - **Output directory**
+     - ``OUTPUT``
+     - [folder]
+     - The output folder for the tiles.
+   * - **CSV file containing the tile(s) georeferencing information**
+     - ``OUTPUT_CSV``
+     - [file]
+     - The output folder for the tiles.
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]: No no...

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst [hunk 10]
```diff
@@ -753,6 +753,190 @@ Outputs
...
+       * Save to Temporary Directory
+       * Save to Directory...
+
+       The file encoding can also be changed here.
+   * - **CSV file containing the tile(s) georeferencing information**
+     - ``OUTPUT_CSV``
+     - [file]
+
+       Default: ``[Skip output]``
+     - Specify the output folder for the tiles.
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]: No no...

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst [hunk 11]
```diff
@@ -753,6 +753,190 @@ Outputs
      - The HTML file containing information about the input raster layer
 
 
+.. _gdalretile:
+
+Retile
+------
+This utility will retile a set of input tile(s).
+All the input tile(s) must be georeferenced in the same
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]:
```suggestion
All the input tiles must be georeferenced in the same
```

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst (hunk 2 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]: Cut and paste from GDAL docs...

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst (hunk 6 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]: Cut and paste...

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst (hunk 5 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]: Same here, but it is cut and paste.

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst (hunk 7 above)
Comment (havatv) [inline source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]:
```suggestion
     - The input raster files
```

Comment (havatv) [reply source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]: No.  Some kind of modified cut and paste, I think.

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst (hunk 8 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]: So it says.

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst (hunk 10 above)
Comment (havatv) [inline source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]:
```suggestion
     - 
```

Comment (havatv) [reply source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]: :-)

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst (hunk 9 above)
Comment (havatv) [inline source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]:
```suggestion
     - The CSV file with georeferencing information for the tiles.
```

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst (hunk 5 above)
Comment (DelazJ) [reply source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]: Yeah, I guessed it was copy and paste (we don't need to reinvent the wheel) but if we don't expose all the features of the original alg, we shouldn't claim that. Even if that can be done with command line options, I'd prefer we stick to what we provide to users, up to advanced gdal techies to do what they know. This is what we usually do with the gdal based algs.

**Comment by havatv** [comment]
Thanks, @DelazJ!

File: source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst (hunk 5 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/gdal/rastermiscellaneous.rst]: Good point.  I have tried to address this in PR #4724.

**Pull request merged**